### PR TITLE
[Merged by Bors] - migrated to clap 3

### LINKED ISSUE #1955: Update CLI to use Clap 3.0
We currently use [`structopt`](https://crates.io/crates/structopt) for our CLI, but now that Clap has released its 3.0 version, the functionality has improved to the point where `structopt` is no longer needed: https://crates.io/crates/clap

We would like to move from using `structopt` to using `clap`.

### PR DESCRIPTION
**Pull request by manthanabc**
This Pull Request fixes/closes #1955.

It changes the following:

- changes structopt to clap

### CODE REVIEW

**Comment by manthanabc** [comment]
i think i need a code review i am confused when to use clap::structopt and clap both seem to do the same thing

File: boa_cli/src/main.rs [hunk 1]
```diff
@@ -64,8 +64,8 @@ use boa_interner::Interner;
 use colored::{Color, Colorize};
 use rustyline::{config::Config, error::ReadlineError, EditMode, Editor};
 use std::{fs::read, io, path::PathBuf};
-use structopt::{clap::arg_enum, StructOpt};
-
+//use structopt::{clap::arg_enum, StructOpt};
```
Comment (Razican) [inline boa_cli/src/main.rs:67]: I guess this is no longer needed, right?

File: boa_cli/src/main.rs [hunk 2]
```diff
@@ -117,6 +117,7 @@ impl Opt {
     }
 }
 
+/*
 arg_enum! {
```
Comment (Razican) [inline boa_cli/src/main.rs:120]: If this is no longer needed, can it be removed?

**Review by Razican — COMMENTED** [review]
Thank you!! Looking very good!

**Comment by Razican** [comment]
I think `clap::structopt` might be a compatibility layer

File: boa_cli/src/main.rs (hunk 2 above)
Comment (manthanabc) [reply boa_cli/src/main.rs:120]: Yeah i think it should be removed.

**Comment by manthanabc** [comment]
ok i removed the extra code

**Comment by manthanabc** [comment]
i might try getting rid of some depreciated function as warned by clippy later tommarow

**Review by Razican — APPROVED** [review]
Looks good from my side, but you need to run `cargo fmt` to format it and pass the format tests :)

**Review by raskad — APPROVED** [review]

**Pull request closed without merging**